service/module.go

```diff
@@ -23,10 +23,19 @@ package service
 import (
 	"fmt"
 	"sync"
-
-	"github.com/uber-go/tally"
 )
 
+// ModuleProvider provides Modules.
+type ModuleProvider interface {
```

Comment:
Yes, we need a separate entity, otherwise the chicken-and-egg problem that lead to modules not knowing their name